api/src/data_migration/setup_foreign_tables.py

```diff
@@ -64,66 +34,39 @@ def setup_foreign_tables(db_session: db.Session) -> None:
     logger.info("Successfully ran setup-foreign-tables")
 
 
-def build_sql(table_name: str, columns: list[Column], is_local: bool, schema_name: str) -> str:
+def build_sql(table: sqlalchemy.schema.Table, is_local: bool, schema_name: str) -> str:
     """
     Build the SQL for creating a possibly foreign data table. If running
     with is_local, it instead creates a regular table.
 
-    Assume you have a table with two columns, an "ID" primary key column, and a "description" text column,
-    you would call this as::
-
-        build_sql("EXAMPLE_TABLE", [Column("ID", "integer", is_nullable=False, is_primary_key=True), Column("DESCRIPTION", "text")], is_local)
-
-    Depending on whether the is_local bool is true or false would give two different outputs.
-
     is_local is True::
 
-        CREATE TABLE IF NOT EXISTS foreign_example_table (ID integer CONSTRAINT EXAMPLE_TABLE_pkey PRIMARY KEY NOT NULL,DESCRIPTION text)
+        CREATE TABLE IF NOT EXISTS foreign_example_table
+        (ID integer CONSTRAINT EXAMPLE_TABLE_pkey PRIMARY KEY NOT NULL,DESCRIPTION text)
 
     is_local is False::
 
-        CREATE FOREIGN TABLE IF NOT EXISTS foreign_example_table (ID integer OPTIONS (key 'true') NOT NULL,DESCRIPTION text) SERVER grants OPTIONS (schema 'EGRANTSADMIN', table 'EXAMPLE_TABLE')
+        CREATE FOREIGN TABLE IF NOT EXISTS foreign_example_table
+        (ID integer OPTIONS (key 'true') NOT NULL,DESCRIPTION text)
+        SERVER grants OPTIONS (schema 'EGRANTSADMIN', table 'EXAMPLE_TABLE')
     """
 
-    column_sql_parts = []
-    for column in columns:
-        column_sql = f"{column.column_name} {column.postgres_type}"
-
-        # Primary keys are defined as constraints in a regular table
-        # and as options in a foreign data table
-        if column.is_primary_key and is_local:
-            column_sql += f" CONSTRAINT {table_name}_pkey PRIMARY KEY"
-        elif column.is_primary_key and not is_local:
-            column_sql += " OPTIONS (key 'true')"
-
-        if not column.is_nullable:
-            column_sql += " NOT NULL"
-
-        column_sql_parts.append(column_sql)
-
-    create_table_command = "CREATE FOREIGN TABLE IF NOT EXISTS"
+    create_table = sqlalchemy.schema.CreateTable(table, if_not_exists=True)
     if is_local:
-        # Don't make a foreign table if running locally
-        create_table_command = "CREATE TABLE IF NOT EXISTS"
-
-    create_command_suffix = (
-        f" SERVER grants OPTIONS (schema 'EGRANTSADMIN', table '{table_name}')"  # noqa: B907
-    )
-    if is_local:
-        # We don't want the config at the end if we're running locally so unset it
-        create_command_suffix = ""
-
-    return f"{create_table_command} {schema_name}.foreign_{table_name.lower()} ({','.join(column_sql_parts)}){create_command_suffix}"
+        compiler = create_table.compile(
+            dialect=sqlalchemy.dialects.postgresql.dialect(),
+            schema_translate_map={"grants": schema_name},
```

Comment:
We should use the `Schema` enum here to make sure any changes to the value work.